pyvista/core/utilities/points.py

```diff
@@ -179,11 +182,450 @@ def lines_from_points(points, close=False):
     return poly
 
 
-def fit_plane_to_points(points, return_meta=False):
-    """Fit a plane to a set of points using the SVD algorithm.
+def principal_axes_transform(points, **kwargs):
+    """Compute the principal axes transform.
 
-    The plane is automatically sized and oriented to fit the extents of
-    the points.
+    This function computes the transformation matrix which will:
+        1. Translate ``points`` such that their centroid is at the origin, then
+        2. Rotate ``points`` to align their principal axes to the XYZ axes.
+
+    See :func:`~pyvista.principal_axes_vectors` for more information and
+    for additional keyword arguments.
+
+    .. versionadded:: 0.43.0
+
+    Notes
+    -----
+    If the transform cannot be computed, the identity matrix is returned.
+
+    Parameters
+    ----------
+    points : array_like[float]
+        Points array. Accepts a single point or several points as a
+        Nx3 array.
+
+    **kwargs : dict, optional
+        Keyword arguments passed to :func:`~pyvista.principal_axes_vectors`.
+
+    Returns
+    -------
+    numpy.ndarray
+        The 4x4 transformation matrix which aligns the points to the XYZ axes
+        at the origin.
+
+    Examples
+    --------
```

Comment:
This section is informative enough to add a stand-alone example script. It would be better to add a script.